### DISCUSSION TITLE
How to create a sub tag form `[^1^]`?

### DISCUSSION THREAD

**Discussion by tumars** [remarkjs/remark#1276] [Q&A]
![134BA37F-6603-476C-8BFD-B07683D6C05E](https://github.com/remarkjs/remark/assets/9925509/1bbf0b61-f101-4bc0-a724-9d38cceb314d)

I hope make a sup tag like this, which used in AI chat context。How to implement this？

The origin markdown text is like that：
> 1. **BBC News** reports that the US aviation regulator has grounded 737 Max 9s for emergency checks after an incident in Oregon[^1^].

the `sup` tag is targeted by `[^1^]`

**Comment by wooorm** [comment]
A what? Please spend time framing your question. I can’t see in your head.
https://github.com/remarkjs/.github/blob/main/support.md
What markdown input. What HTML output. Etc etc etc etc.

**Reply by tumars** [reply]
Apologize for my poor expressive skills.
Here is what I wanna:

**INPUT**
markdown:
```text
BBC News reports that the US aviation regulator has grounded 737 Max 9s for emergency checks after an incident in Oregon[^1^].
```

and a extra json: 
```json
{
  "1": "http://www.xxx.com"
}
```

**OUTPUT**

```html
<a href="https://www.xxx.com" class="customClass">BBC News reports that the US aviation regulator has grounded 737 Max 9s for emergency checks after an incident in Oregon</a>
<a href="https://www.xxx.com" class="customClass"><sup class="citation-sup">1</sup></a>
```

----

But I find this Q&A: https://github.com/orgs/remarkjs/discussions/1270
May I can use the same method as it does.

**Reply by ChristianMurphy** [reply]
@tumars welcome :wave:
The short answer is that doesn't exist.
The closest alternative is footnote syntax from GFM https://github.com/remarkjs/remark-gfm

I'd recommend using existing syntax over creating your own flavor of markdown.
You'd have to document your flavor and train your AI and any human authors in how to write your custom flavor of markdown. https://github.com/micromark/micromark#extending-markdown

If you do choose to create your own flavor create a plugin to do so, there is more information on plugins here: https://unifiedjs.com/learn/
